README.md

```diff
@@ -35,48 +35,60 @@
-Whether you want to perform question answering (QA) or semantic document search, you can use the state-of-the-art NLP models in Haystack to provide unique search experiences and allow your users to query in natural language.
-Haystack is built in a modular fashion so that you can combine the best technology from other open source projects, like Hugging Face's transformers, Elasticsearch, or Milvus.
+[Haystack](https://haystack.deepset.ai/) is an end-to-end NLP framework that enables you to build NLP applications powered by LLMs, Transformer models, vector search and more. Whether you want to perform question answering, answer generation, semantic document search, or build tools that are capable of complex decision making and query resolution, you can use the state-of-the-art NLP models with Haystack to build end-to-end NLP applications solving your use case
 
-<p align="center"><img src="https://raw.githubusercontent.com/deepset-ai/haystack/main/docs/img/main_example.gif"></p>
+## Core Concepts
+
+🏃‍♀️ **[Pipelines](https://docs.haystack.deepset.ai/docs/pipelines):** This is the standard Haystack structure that can connect to your data and perform an NLP task on it, that you define. ****A Pipeline is a structure where data flows from one Node to the next where you define how Nodes interact with each other, and how one Node pushes data to the next.
+
+An example pipeline would consist of one `Retriever` Node and one `Reader` Node. When the pipeline is run with a query, the Retriever will first retrieve the relevant documents to the query, and then, the Reader will extract a final answer.
+
+⚛️ **[Nodes](https://docs.haystack.deepset.ai/docs/nodes_overview):** Each node achieves one thing. Such as preprocessing documents, retrieving documents, using language models to answer questions and so on.
+
+🕵️ **[Agent](https://docs.haystack.deepset.ai/v1.15/docs/agent):** (since of v1.15) An Agent is a component that is powered by an LLM, such as GPT-3. It is able to make decisions on what the next best course of action is so as to get to the result of a query. It does so by making use of Tools that are available to it. An Agent is a higher level component than a pipeline. Where a pipeline has a clear start and end, an Agent is able to decide whether the query has resolved or not. It may also make use of a Pipeline that can do something really well, as a Tool.
+
+🛠️ **[Tools](https://docs.haystack.deepset.ai/v1.15/docs/agent#tools):** You can think of a Tool as an expert, that is able to do something really well. Such as a calculator, good at mathematics. Or a [WebRetriever](https://docs.haystack.deepset.ai/v1.15/docs/agent#web-tools), good at retrieving pages from the internet. A Node or pipeline in Haystack can also be used as a Tool. A Tool is a component that is used by an Agent, to resolve complex queries.
```

Comment:
Same comment about the docs URL -> when 1.15 is published, it will no longer appear in the URL. 

"..expert, that is able to do something.." -> "...expert that can do something" (no comma)

"A Tool is a component that is used by an Agent" -> An Agent uses Tools to resolve complex queries. (I don't think it's always "a component" in the way we understand Haystack components)